Added an article about row-oriented operations with purrr

chsafouane opened the pull request:
Issue 610: row-oriented doc

hadley commented:
Fixes #610

hadley commented on .Rbuildignore:
```diff
@@ -18,3 +18,4 @@
 ^\.github$
 ^TAGS$
 ^\.httr-oauth$
+^vignettes/runif-via-pmap\.Rmd$
```
Why are you build-ignoring this vignette?

chsafouane replied:
I'm adding it as an article and not as vignette. I would like to be added to purrr website

hadley replied:
Is there some reason it shouldn't be a vignette?

chsafouane replied:
I turned it into a vignette but there was a comment by Jenny that this issue of iterating over rows shouldn't be made a vignette because it relates to many packages https://github.com/tidyverse/purrr/issues/610#issuecomment-451248219 but I think this vignette is just about purrr

hadley commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,213 @@
+The first step is to store triples in a dataframe and then iterate over rows.
+
+```{r include=FALSE}
+library(tidyverse)
````
I think this should just be `library(purrr)`

hadley commented on vignettes/runif-via-pmap.Rmd:
```diff
@@ -0,0 +1,213 @@
+---
+title: "Generate data from different distributions via pmap()"
```
This title feels a bit too specific — can you make it more general?

hadley commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,213 @@
+```{r setup}
+library(purrr)
+```
+
````
I think you need a paragraph that motivate the vignette. Why should someone read it? What will they learn?

chsafouane replied:
Added at the top of the vignette

hadley commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,213 @@
+
+```
+
+Now, let us try to do the same thing with `pmap()`.
````
```suggestion
Now, let us try to do the same thing with `pmap()`:
```

hadley commented on vignettes/runif-via-pmap.Rmd:
```diff
@@ -0,0 +1,213 @@
+
+### Finessing variable and argument names
+
+It is not mandatory for variable names to be the same as the argument names of the function. This can be arranged on the fly.In the example below, we do exactly this.
```
This example feels a bit forced. Is there a more natural scenario in which renaming the variables is needed? Maybe you could simulate from a different distribution?

hadley commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,213 @@
+One can also create a wrapper around `runif()` that does the correspondance between column names and argument names on the fly.
+
+```{r}
+# 1st option of the wrapper
````
I would not do this, personally.

hadley commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,213 @@
+pmap(foofy, my_runif)
+```
+ 
+A third option that comes in handy is to refer to inputs by their position `..i`. Yet, one should be careful when inputting the column numbers
````
I wouldn't do this either.

hadley commented on vignettes/runif-via-pmap.Rmd:
```diff
@@ -0,0 +1,213 @@
+# OR
+
+set.seed(123)
+df_oops %>%
```
I think you should include this example, but not run it.

chsafouane replied:
Done

hadley commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,213 @@
+  pmap(runif)
+```
+
+A custom wrapper with the dot syntax `...` can also get the job done.
````
... that uses `...` to ignore additional argument can also ...

hadley commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,213 @@
+
+```{r}
+set.seed(123)
+(df_aug <- df %>%
````
If you do this, I think you need to explain that the `.` comes from magrittr, so it will only work if in a pipe, and it will probably give surprising results if `df` is grouped.

chsafouane replied:
Done

hadley commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,213 @@
+  mutate(data = pmap(select(., -oops), runif))
+```
+
+### Review
````
I think this would make sense at the beginning of the doc to give an overview of what will be covered.

chsafouane replied:
Made some fixes

lionel- commented:
Is this good to merge for the point release @hadley?

lionel- commented:
@chsafouane, since the examples are taken from https://github.com/jennybc/row-oriented-workflows, could you include Jenny as co-author please?

chsafouane commented:
I've added Jenny as a co-author. I think we might add a link to her repo and cite it in the vignette as I've only turned her repo to a vignette (little formatting and changed some of the examples)

lionel- commented:
Agreed, a link and a mention would be great.

chsafouane commented:
I've just added a reference to Jenny's repo

chsafouane commented on vignettes/runif-via-pmap.Rmd:
````diff
@@ -0,0 +1,190 @@
+```
+
+```{r include=F}
+library(dplyr)
````
Honestly, I don't what why I did this... I've combined the two chunks

hadley commented:
I now think this is out of scope for purrrr, since it falls an the documentation space that lies in between multiple packages. We don't currently have a good home for such docs, but @batpigandme is working on it, and this work is likely to make a future appearance elsewhere.
